### DISCUSSION TITLE
How to plot specific orientations (show markers) on pole figures?

### DISCUSSION THREAD

**Discussion by masayoshi-kumagai** [mtex-toolbox/mtex#2246] [Ask Anything]
I'm having trouble indicating plots with specific hkl directions on pole figures. I would appreciate your advice on the following details problem. At first, I showed three pole figures with (111), (200) and (220). After that, I tried indicating marks for [111], [100] and [110] directions using the code below. However, the code didn't work and showed six figures...
```matlab
ori111 = Miller(1, 1, 1, odf.CS);
ori110 = Miller(1, 1, 0, odf.CS);
ori100 = Miller(1, 0, 0, odf.CS);

figure;
plotPDF(odf, h, 'grid', 'grid_res', 15*degree, 'colorrange', 'equal', 'antipodal');
subplot(1, 3, 1);
hold on; 
plot(ori111, 'Marker', 'o', 'MarkerSize', 8, 'MarkerFaceColor', 'r', 'MarkerEdgeColor', 'k');
plot(ori110, 'Marker', 's', 'MarkerSize', 8, 'MarkerFaceColor', 'b', 'MarkerEdgeColor', 'k');
plot(ori100, 'Marker', '^', 'MarkerSize', 8, 'MarkerFaceColor', 'g', 'MarkerEdgeColor', 'k');
hold off;

subplot(1, 3, 2);
plotPDF(odf, h, 'grid', 'grid_res', 15*degree, 'colorrange', 'equal', 'antipodal');
hold on; 
plot(ori111, 'Marker', 'o', 'MarkerSize', 8, 'MarkerFaceColor', 'r', 'MarkerEdgeColor', 'k');
plot(ori110, 'Marker', 's', 'MarkerSize', 8, 'MarkerFaceColor', 'b', 'MarkerEdgeColor', 'k');
plot(ori100, 'Marker', '^', 'MarkerSize', 8, 'MarkerFaceColor', 'g', 'MarkerEdgeColor', 'k');
hold off;

subplot(1, 3, 3);
plotPDF(odf, h, 'grid', 'grid_res', 15*degree, 'colorrange', 'equal', 'antipodal');
hold on; 
plot(ori111, 'Marker', 'o', 'MarkerSize', 8, 'MarkerFaceColor', 'r', 'MarkerEdgeColor', 'k');
plot(ori110, 'Marker', 's', 'MarkerSize', 8, 'MarkerFaceColor', 'b', 'MarkerEdgeColor', 'k');
plot(ori100, 'Marker', '^', 'MarkerSize', 8, 'MarkerFaceColor', 'g', 'MarkerEdgeColor', 'k');
hold off;

mtexColorbar;
```
![PDFwithMarks](https://github.com/user-attachments/assets/0eab0438-0a44-4234-b1dc-ccc3927ad0be)

**Comment by ralfHielscher** [comment]
There is a couple of problems with this code:

1. `h` is not defined and should be of type Miller
2. `ori` is not an orientation but just a crystal direction
3. you should not use `subplot` but `nextAxis`

Here is a code that might give what you want
```matlab
% the crystal directions of the pole figures you want to plot
h = Miller({1,1,1},{1,1,0},{1,0,0},odf.CS)

plotPDF(odf, h, 'grid', 'grid_res', 15*degree, 'colorrange', 'equal', 'antipodal');
mtexColorbar;

% the orientation you want to annotate
ori = orientation.rand(odf.CS)

plot(ori,'add2all','MarkerColor','red')
```
Ralf.

**Reply by masayoshi-kumagai** [reply]
Dear Ralf,
Thanks your helpful advise. I was able to add any plots using "plot(ori,'add2all','MarkerColor','red')" However, I still lack some knowledge about the concept of "ori". I would appreciate it if you could let me know how to define the "ori" for multiple figures, e.g. {1 1 1} on each pole figure of (1 1 1), (1 10) and (1 1 0).
Kind regards,
MK

**Comment by ralfHielscher** [comment]
I'm afraid you need to read a bit more what an orientation actually is. It is an description of how a crystal reference frame is aligned to a specimen reference frame, see e.g. https://mtex-toolbox.github.io/DefinitionAsCoordinateTransform.html.
Orientations can be defined in numerous way, see e.g. https://mtex-toolbox.github.io/OrientationDefinition.html.
Pole figures are ways of visualizing orientations. So one can not say to define and orientation for a pole figure.
One can define an orientation that appears at a specific spot in one pole figure and on another spot in a second pole figure. This can be accomplished by the command [`orientation.map`](https://mtex-toolbox.github.io/orientation.map.html)
Ralf

**Comment by masayoshi-kumagai** [comment]
Dear Ralf,
Thank you for your support. I will check the references.

Kind regards,
MK
________________________________
差出人: Ralf Hielscher ***@***.***>
送信日時: Monday, October 14, 2024 4:47:45 PM
宛先: mtex-toolbox/mtex ***@***.***>
CC: KUMAGAI Masayoshi ***@***.***>; Manual ***@***.***>
件名: Re: [mtex-toolbox/mtex] How to plot specific orientations (show markers) on pole figures? (Discussion #2246)


I'm afraid you need to read a bit more what an orientation actually is. It is an description of how a crystal reference frame is aligned to a specimen reference frame, see e.g. https://mtex-toolbox.github.io/DefinitionAsCoordinateTransform.html.
Orientations can be defined in numerous way, see e.g. https://mtex-toolbox.github.io/OrientationDefinition.html.
Pole figures are ways of visualizing orientations. So one can not say to define and orientation for a pole figure.
One can define an orientation that appears at a specific spot in one pole figure and on another spot in a second pole figure. This can be accomplished by the command orientation.map<https://mtex-toolbox.github.io/orientation.map.html>
Ralf
